internal/provider/incident_catalog_entry_resource.go

```diff
@@ -296,9 +329,88 @@ func (r *IncidentCatalogEntryResource) ImportState(ctx context.Context, req reso
 	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
 }
 
-func (r *IncidentCatalogEntryResource) buildModel(entry client.CatalogEntryV2) *IncidentCatalogEntryResourceModel {
+func (r *IncidentCatalogEntryResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	var data IncidentCatalogEntryResourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// If managed_attributes is not set, all attributes are valid
+	if data.ManagedAttributes.IsNull() || data.ManagedAttributes.IsUnknown() {
+		return
+	}
+
+	// Extract the managed attribute IDs
+	managedAttributesMap := map[string]bool{}
+	for _, attrIDElem := range data.ManagedAttributes.Elements() {
+		if attrIDElem.IsUnknown() {
+			continue
+		}
+
+		attrIDStr, ok := attrIDElem.(types.String)
+		if !ok {
+			continue
+		}
+
+		managedAttributesMap[attrIDStr.ValueString()] = true
+	}
+
+	// Check that each attribute in attribute_values is managed
+	for idx, attributeValue := range data.AttributeValues {
+		if attributeValue.Attribute.IsUnknown() {
+			// Ideally we would look at the reference for this, but the framework
+			// doesn't seem to expose this, so we just have to skip the check.
+			continue
+		}
+
+		attributeID := attributeValue.Attribute.ValueString()
+		if !managedAttributesMap[attributeID] {
+			resp.Diagnostics.AddAttributeError(
+				path.Root("attribute_values").AtListIndex(idx),
+				"Unmanaged Attribute",
+				fmt.Sprintf("Attribute ID %q is specified in attribute_values but is not in the managed_attributes set. "+
```

Comment:
Should we not only say this for schema_only attributes?

Reply:
In this resource, we don't actually know whether the attribute is schema-only or not: we just have its ID :(